src/main/menu/MainMenu.java

```diff
@@ -1,30 +1,119 @@
 package main.menu;
 
 import javax.swing.JPanel;
+
+import main.game_engine.PlayPanel;
+import main.menu.leaderboard.LeaderboardPanel;
+import main.menu.shop.ShopGUI;
+import main.utilities.CommonMethods;
+import main.utilities.Constants;
+import main.utilities.Constants.PANEL;
+import main.utilities.GameSettings;
+
 import java.awt.CardLayout;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
 
+/**
+ * MainMenu is the container for CardLayout
+ */
 public class MainMenu extends JPanel implements Menu {
 
 	public static final CardLayout cardLayout = new CardLayout();
+	private final MenuPanel menuPanel;
+	private final LeaderboardPanel leaderboardPanel;
+	private final PlayPanel playPanel;
+	private final ShopGUI shopGUI;
+	private final EOGMenuGUI EOGMenuGUI;
+	private final Tutorial tutorial;
+	private final SkinSelectionPanel selectionPanel;
+
+	/**
+	 * @param gameSettings - GameSettings
+	 */
+	public MainMenu(GameSettings gameSettings) {
+
+		menuPanel = new MenuPanel(this);

```

Comment:
Panel initializations must be made after checking the existence of the savings file to prevent IOException.

Reply:
done